central/reprocessor/reprocessor.go

```diff
@@ -339,6 +339,17 @@ func (l *loopImpl) reprocessImage(id string, fetchOpt imageEnricher.FetchOption,
 				logging.ImageName(image.GetName().GetFullName()), logging.Err(err))
 			return nil, false
 		}
+		// We need to fetch the image again to make sure all fields are populated.
+		// GetImage will internally call a Merge function which will use the CVEEdges table to enrich fields like
+		// FirstImageOccurrence and FirstSystemOccurrence.
+		image, exists, err = l.images.GetImage(allAccessCtx, id)
+		if err != nil {
+			log.Errorw("Error fetching image from database", logging.ImageID(id), logging.Err(err))
+			return nil, false
+		}
+		if !exists {
```

Comment:
This is a logical error, right? We first upserted the image and now we can't get it. I think we shall at least log such an occurrence.

Reply:
Good point! I'll add a log here.

Reply:
I have a problem with that. We upsert then get to have some side effect. 
Can we achieve it in different way

Reply:
> Can we achieve it in different way

The logic to enrich those fields is embedded in the store. I think refactoring that is going to be a pretty big change so I'd vote to just re-fetch the image to trigger the logic for now.